### PR TITLE
feat: create notification schema

### PR DESCRIPTION
**Pull request by idoshamun**
Add 3 new tables notification, notification avatar and notification attachments. Add GraphQL types accordingly and GraphORM proper sorting for attachments and avatars when querying for notification

WT-774 #done

### CODE REVIEW

File: src/entity/Notification.ts [hunk 1]
```diff
@@ -1,18 +1,81 @@
...
+  OneToMany,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { User } from './User';
+import { NotificationAvatar } from './NotificationAvatar';
+import { NotificationAttachment } from './NotificationAttachment';
+
+export type NotificationType =
+  | 'community_picks_failed'
+  | 'community_picks_success'
```
Comment (rebelchris) [inline src/entity/Notification.ts:15]: Should this not be `community_picks_succeeded` to be inline with correct tense?

File: src/entity/NotificationAttachment.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Column, Entity, Index, ManyToOne, PrimaryColumn } from 'typeorm';
+import { Notification } from './Notification';
+
+export type NotificationAttachmentType = 'post';
```
Comment (rebelchris) [inline src/entity/NotificationAttachment.ts:4]: I'm actually thinking now, would we not want to opt to store the referenced post as well?
In the case of the post being deleted for instance?

File: src/entity/NotificationAvatar.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { Column, Entity, Index, ManyToOne, PrimaryColumn } from 'typeorm';
+import { Notification } from './Notification';
+
+export type NotificationAvatarType = 'source' | 'user';
```
Comment (rebelchris) [inline src/entity/NotificationAvatar.ts:4]: Same question here, for instance with spammers who get deleted and perhaps use inappropriate images?

**Review by rebelchris — APPROVED** [review]
Happy with the code, just have some additional questions around how we safe things.

File: src/entity/Notification.ts (hunk 1 above)
Comment (idoshamun) [reply src/entity/Notification.ts:15]: Copied from the DR. I'll fix that!

File: src/entity/NotificationAttachment.ts (hunk 2 above)
Comment (idoshamun) [reply src/entity/NotificationAttachment.ts:4]: Good point!

File: src/entity/NotificationAvatar.ts (hunk 3 above)
Comment (idoshamun) [reply src/entity/NotificationAvatar.ts:4]: Here we already store the `referenceId`

**Pull request merged**